File: bms/go/geo/geo_r3_Vector_Distance/geo_r3_Vector_Distance.go
package test

import (
	"math"
)

var ()

const ()

func (v geo_r3_Vector) Distance(ov geo_r3_Vector) float64 { return v.Sub(ov).Norm() }

func (v geo_r3_Vector) Sub(ov geo_r3_Vector) geo_r3_Vector {
	return geo_r3_Vector{v.X - ov.X, v.Y - ov.Y, v.Z - ov.Z}
}

func (v geo_r3_Vector) Norm() float64 { return math.Sqrt(v.Dot(v)) }

func (v geo_r3_Vector) Dot(ov geo_r3_Vector) float64 {
	return float64(v.X*ov.X) + float64(v.Y*ov.Y) + float64(v.Z*ov.Z)
}

type geo_r3_Vector struct {
	X, Y, Z float64
}

File: translations_only/fail/go/geo/geo_r3_Vector_Distance/mistral/Restart/geo_r3_Vector_Distance.rs


use std::f64;

type geo_r3_Vector = Vector;

struct Vector {
    x: f64,
    y: f64,
    z: f64,
}

impl Vector {
    fn sub(&self, ov: Vector) -> Vector {
        Vector {
            x: self.x - ov.x,
            y: self.y - ov.y,
            z: self.z - ov.z,
        }
    }

    fn dot(&self, ov: &Vector) -> f64 {
        self.x * ov.x + self.y * ov.y + self.z * ov.z
    }

    fn norm(&self) -> f64 {
        self.dot(self).sqrt()
    }
}

fn distance(v: Vector, ov: Vector) -> f64 {
    let sub = v.sub(ov);
    sub.norm()
}

